src/web/page.rs

```diff
@@ -119,42 +111,145 @@ impl<T: ToJson> Page<T> {
+        let mut varsb = BTreeMap::new();
+        varsb.insert("test2".into(), true);
+        let mut varsi = BTreeMap::new();
+        varsi.insert("test3".into(), 1337);
+
+        let page = Page {
+            title: None,
+            content: vec![release.clone()],
+            status: status::Status::Ok,
+            varss,
+            varsb,
+            varsi,
+            rustc_resource_suffix: &*RUSTC_RESOURCE_SUFFIX,
+        };
+
+        let correct_json = json!({
+            "content": [{
+                "name": "lasso",
+                "version": "0.1.0",
+                "description": null,
+                "target_name": null,
+                "rustdoc_status": false,
+                "release_time": super::super::duration_to_str(time),
+                "release_time_rfc3339": time::at(time).rfc3339().to_string(),
+                "stars": 0
+            }],
+            "varss": { "test": "works" },
+            "varsb": { "test2": true },
+            "varsi": { "test3": 1337 },
+            "rustc_resource_suffix": &*RUSTC_RESOURCE_SUFFIX,
+            "cratesfyi_version": crate::BUILD_VERSION,
+            "cratesfyi_version_safe": crate::BUILD_VERSION
+                .replace(" ", "-")
+                .replace("(", "")
+                .replace(")", ""),
+            "has_global_alert": crate::GLOBAL_ALERT.is_some()
+        });
+
+        assert_eq!(correct_json, serde_json::to_value(&page).unwrap());
+    }
+
+    #[test]
+    fn load_page_from_releases() {
+        crate::test::wrapper(|env| {
+            let db = env.db();
+            db.fake_release().name("foo").version("0.1.0").create()?;
+            let packages = releases::get_releases(&db.conn(), 1, 1, releases::Order::ReleaseTime);
+
+            let mut varsb = BTreeMap::new();
+            varsb.insert("show_search_form".into(), true);
+            varsb.insert("hide_package_navigation".into(), true);
+
+            let correct_page = Page {
+                title: None,
+                content: packages.clone(),
+                status: status::Status::Ok,
+                varss: BTreeMap::new(),
+                varsb,
+                varsi: BTreeMap::new(),
+                rustc_resource_suffix: &RUSTC_RESOURCE_SUFFIX,
+            };
+
+            let page = Page::new(packages)
+                .set_true("show_search_form")
+                .set_true("hide_package_navigation");
+
+            assert_eq!(page, correct_page);
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn serialize_global_alert() {
+        let alert = GlobalAlert {
+            url: "http://www.hasthelargehadroncolliderdestroyedtheworldyet.com/",
+            text: "THE WORLD IS ENDING",
+            css_class: "THE END IS NEAR",
+            fa_icon: "https://gph.is/1uOvmqR",
+        };
+
+        let correct_json = json!({
+            "url": "http://www.hasthelargehadroncolliderdestroyedtheworldyet.com/",
+            "text": "THE WORLD IS ENDING",
```

Comment:
This is clearly incorrect.
```suggestion
            text: "THE WORLD WILL SOON END",
            css_class: "THE END IS NEAR",
            fa_icon: "https://gph.is/1uOvmqR",
        };

        let correct_json = json!({
            "url": "http://www.hasthelargehadroncolliderdestroyedtheworldyet.com/",
            "text": "THE WORLD WILL SOON END",
```

Reply:
I donno, I think the first may be pretty accurate as of now